client/state/themes/reducer.js

```diff
@@ -346,6 +349,51 @@ export const lastQuery = createReducer( {}, {
 	} )
 } );
 
+/**
+ * Returns the updated previewing theme state
+ * The state reflects a mapping of themId that can be used to show its preview.
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Object}        Updated state
+ */
+export const previewingTheme = createReducer( {}, {
+	[ PREVIEWING_THEME_DATA ]: ( state, { themeData } ) => ( {
+		...state,
+		themeData
+	} )
+} );
+
+/**
+ * Returns the updated previewing theme state
+ * The state reflects a mapping of themId that can be used to show its preview.
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Object}        Updated state
+ */
+export const previewingThemeOptions = createReducer( {}, {
+	[ PREVIEWING_THEME_OPTIONS ]: ( state, { themeOptions } ) => ( {
+		...state,
+		themeOptions
+	} )
+} );
+
+/**
+ * Returns the updated previewing theme state
+ * The state reflects a mapping of themId that can be used to show its preview.
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Object}        Updated state
+ */
+export const previewingThemeState = createReducer( {}, {
+	[ THEME_PREVIEW_STATE ]: ( state, { isEnabled } ) => ( {
+		...state,
+		isEnabled
```

Comment:
I think this is unnecessarily adding an extra object attr level -- `state` could just store the value of `isEnabled` instead of storing an object with an `isEnabled` attr, i.e.

```js
[ THEME_PREVIEW_STATE ]: ( state, { isEnabled } ) => ( isEnabled )
```